### PR TITLE
Deletes unimplemented vNEXT doc.

### PR DESCRIPTION
**Pull request by mathetake**
As a part of the project overhaul following the discussion in https://github.com/proxy-wasm/spec/pull/38#issuecomment-1577831426, 
this commit deletes the never-implemented vNEXT ABI documentation 
which has caused confusion to users and those willing to contribute.

The latest ["implemented" specification version is `v0.2.1`](https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/master/src/wasm.cc#L128-L131), and the proper 
documentation of v0.2.1 will be added to this repository as a follow-up PR.

### CODE REVIEW

**Comment by mathetake** [comment]
cc @mpwarres @martijneken @vikaschoudhary16 @jcchavezs

**Comment by jcchavezs** [comment]
Good to see some movement here!

**Review by mpwarres — APPROVED** [review]
LGTM. Would like to get @PiotrSikora 's ok on this as well.

**Comment by jcchavezs** [comment]
Ping @PiotrSikora

**Comment by mathetake** [comment]
I believe the agreement is already made between @PiotrSikora and @mpwarres as-in https://github.com/proxy-wasm/spec/pull/38#issuecomment-1577831426, so I will go ahead and merge this. Thanks!

**Comment by PiotrSikora** [comment]
This is wrong and shouldn't have been merged.

> I believe the agreement is already made between @PiotrSikora and @mpwarres as-in [#38 (comment)](https://github.com/proxy-wasm/spec/pull/38#issuecomment-1577831426), so I will go ahead and merge this. Thanks!

@mpwarres clearly asked for my approval on this PR in https://github.com/proxy-wasm/spec/pull/40#pullrequestreview-1468931815, so I don't understand why you chose to "believe" otherwise.

**Comment by codefromthecrypt** [comment]
not sure @mathetake [replied](https://github.com/proxy-wasm/spec/issues/41#issuecomment-1588235039) to the correct issue on vNEXT or not. 

I routinely point people to the [issue that says it is broken](https://github.com/proxy-wasm/spec/issues/36). No one is looking to cause any problems, but we do want to avoid pain. Not two weeks ago, Takeshi and I had to privately explain why this document is confusing to an employee at a company we don't work at (redacted to protect the innocent)

Screenshot 2023-06-13 at 07 57 24

I'm not trying to convince you that deleting is the right answer, but it certainly wasn't motivated for any reason except to stop the constant damage to our time.

All this said, usually when I'm in an OSS project and someone with access merged something I don't agree with.. If I feel strongly, I'll revert it. I think if you want to revert it, even if that position is unpopular, others would accept it.

So, TL;DR; is my unsolicited 2p as a consumer not a committer, is revert if you feel this was wrong!

**Comment by PiotrSikora** [comment]
> I routinely point people to the [issue that says it is broken](https://github.com/proxy-wasm/spec/issues/36). No one is looking to cause any problems, but we do want to avoid pain. Not two weeks ago, Takeshi and I had to privately explain why this document is confusing to an employee at a company we don't work at (redacted to protect the innocent)
> 
> Screenshot 2023-06-13 at 07 57 24
> I'm not trying to convince you that deleting is the right answer, but it certainly wasn't motivated for any reason except to stop the constant damage to our time.

It says `vNEXT` right there in the link... but why are they looking at the ABI spec in the first place? I feel like there is some context missing.
